src/classes/UTIL_Describe.cls

```diff
@@ -496,6 +498,54 @@ public class UTIL_Describe {
 
         return picklistValues;
     }
+    
+    /*********************************************************************************************************
+     * @description Returns a map of active, user-available fields labels and API name for a given
+     * SObjectType and field type, keyed by  label with a value of api name.
+     * @param objectName Name of the object to retrieve values for.
+     * @param type data type of the fields to retrieve values for.
+     * @return Map<String, String> Map of labels to api name for the object and field type specified.
+     *****/
+    @AuraEnabled
+    public static Map<String, String> getFieldsOfTypeAura(String objectName, String type) {
+		
+        //API name and Schema of field
+        Map<String, Schema.DescribeFieldResult> allFields = UTIL_Describe.getAllFieldsDescribe( objectName );
+        Map<String, String> fields = new Map<String, String>();
+        for(String fieldAPI: allFields.keySet()) {
+            Schema.DescribeFieldResult field = allFields.get(fieldAPI);
+     
+            if (String.valueOf( field.getType() ) == type) {
+                 fields.put(field.getLabel(), fieldAPI);
+            }
+        }
+	
+        return fields;
+    }
+    
+
+    /*********************************************************************************************************
+     * @description Returns field API name for a given object and field label
+     * @param objectName Name of the object to retrieve values for.
+     * @param fieldLabel Label of the field
+     * @return String API Name of the field retrieved
+     *****/
+
+    public static String getFieldAPIName (String objectName, String fieldLabel) {
+    SObjectType objType = Schema.getGlobalDescribe().get(objectName);
```

Comment:
I know this method is currently only called once and not in a loop, but Schema.getGlobalDescribe() is a very expensive call in terms of CPU time. We should do something similar to lines 59-62 in this class and use the cached value if there is one.